trin-storage/src/versioned/id_indexed_v1/store.rs

```diff
@@ -275,12 +276,13 @@ impl<TContentKey: OverlayContentKey> IdIndexedV1Store<TContentKey> {
         self.usage_stats.total_entry_size_bytes += content_size as u64;
         self.usage_stats.report_metrics(&self.metrics);
 
+        let mut dropped_content: Vec<(TContentKey, Vec<u8>)> = Vec::new();
```

Comment:
nit: I don't think that you need to declare type here, so simple `let mut dropped_content = vec![];` should be enough.

Also, you can do something like this:
```rust
let mut dropped_content = if self.pruning_strategy.should_prune(&self.usage_stats) {
    self.prune()?
} else {
    vec![]
}
```